MentorMatchmaker/EmailGeneration.fs

```diff
@@ -0,0 +1,149 @@
+﻿
+[<RequireQualifiedAccess>]
+module MentorMatchmaker.EmailGeneration
+
+open MentorMatchmaker
+open System.Net.Mail
+
+[<AutoOpen>]
+module private Implementation =
+    let LengthOfMentorshipInWeeks = 8 // TODO: I need a a way to retrieve it and to make it not static....
+
+    let rec topicName topic =
+        match topic with
+        | IntroductionToFSharp -> "Introduction to F#"
+        | DeepDiveInFSharp -> "Deep dive in F#"
+        | ContributeToOpenSource -> "Contribute to open-source"
+        | ContributeToCompiler -> "Contribute to compiler"
+        | WebDevelopment -> "Web development"
+        | DistributedSystems -> "Distributed systems"
+        | DomainModelling -> "Domain modeling"
+        | MobileDevelopment -> "Mobile development"
+        | MachineLearning -> "Machine learning"
+        | UpForAnything -> "I am up for anything"
+        | DesigningWithTypes -> "Designing with types"
+        | MetaProgramming -> "Meta programming"
+
+    let printLocalMeetingTimes offset (meetingTimes: Set<WeekTime>) =
+        meetingTimes
+        |> Set.map (fun time -> time.AddHours(offset))
+        |> toWeekTimeRanges
+        |> Seq.groupBy (fun weekTimeRange -> weekTimeRange.Start.Weekday)
+        |> Seq.map (fun (weekday, weekTimeRanges) ->
+            let aggregatedTimes =
+                weekTimeRanges
+                |> Seq.map (fun range -> range.Start.Time.ToShortTimeString() + "-" + range.End.Time.AddHours(1).ToShortTimeString())
+                |> String.concat ", "
+
+            $"{weekday}: {aggregatedTimes}<br>"
+        )
+        |> String.concat("\t\t\t")
```

Comment:
I feel this function is sort of duplicated in Program.fs?

Reply:
Sort of? It's kind of string aggregation whether you do a CLI output or email and they look clearer in their own context I thought. Should we always make them the same?

There could be a case here for moving the CLI output logic into a different module than the main program, but it was just small and localized enough I wasn't convinced.

Reply:
Since it's only for 2 contexts, we're fine here. If we had more to support (doubtful but you never know) there might be a case to be made for a general function they can each hit whenever they need it